src/caf/core/segmentation.py

```diff
@@ -59,23 +59,25 @@ class SegmentationInput(BaseConfig):
     """
 
     enum_segments: list[SegmentsSuper]
-    naming_order: list[str]
-    subsets: dict[str, list[int]] = pydantic.Field(default_factory=dict)
     custom_segments: list[Segment] = pydantic.Field(default_factory=list)
+    subsets: dict[str, list[int]] = pydantic.Field(default_factory=dict)
+    naming_order: list[str]
 
-    @pydantic.validator("subsets", always=True)
+    @pydantic.field_validator("subsets", check_fields=True)
```

Comment:
Yep as mentioned this is done in ''functionality_for_tem" branch. If you're in a pydantic frame of mind you can have a look there, but it will go to you for review when done anyway